model/channel.go

```diff
@@ -126,10 +133,42 @@ func (o *Channel) ExtraUpdated() {
 	o.ExtraUpdateAt = GetMillis()
 }
 
+func (o *Channel) IsGroupOrDirect() bool {
+	return o.Type == CHANNEL_DIRECT || o.Type == CHANNEL_GROUP
+}
+
 func GetDMNameFromIds(userId1, userId2 string) string {
 	if userId1 > userId2 {
 		return userId2 + "__" + userId1
 	} else {
 		return userId1 + "__" + userId2
 	}
 }
+
+func GetGroupDisplayNameFromUsers(users []*User, truncate bool) string {
+	usernames := make([]string, len(users))
+	for index, user := range users {
+		usernames[index] = user.Username
+	}
+
+	sort.Strings(usernames)
+
+	name := strings.Join(usernames, ", ")
+
+	if truncate && len(name) > CHANNEL_NAME_MAX_LENGTH {
+		name = name[:CHANNEL_NAME_MAX_LENGTH]
+	}
+
+	return name
+}
+
+func GetGroupNameFromUserIds(userIds []string) string {
+	sort.Strings(userIds)
+
+	h := sha1.New()
```

Comment:
The hash here is also based off of user IDs already in the system so it's incredibly difficult to manipulate this anyway